### PR TITLE
 Add Single Tag Page With Deep Link Support

### LINKED ISSUE #12: Add Ability to Click Tags
Clicking a tag on a question should open a new page with questions containing the same tag.

`/2.2/questions?order=desc&sort=activity&tagged=android&site=stackoverflow`

### PR DESCRIPTION
**Pull request by tylerbwong**
Closes #12, #22

### CODE REVIEW

File: app/src/main/java/me/tylerbwong/stack/data/DeepLinker.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import me.tylerbwong.stack.ui.questions.detail.QuestionDetailActivity
+import me.tylerbwong.stack.ui.questions.tags.SingleTagQuestionsActivity
+
+object DeepLinker {
+
+    fun resolvePath(context: Context, uri: Uri): Intent? {
```
Comment (brittanyberlanga) [inline app/src/main/java/me/tylerbwong/stack/data/DeepLinker.kt]: Can you add comment explaining the uri formats we expect to handle here?

File: app/src/main/java/me/tylerbwong/stack/ui/questions/tags/SingleTagQuestionsActivity.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+        if (tag.isBlank()) {
+            Toast.makeText(this, getString(R.string.single_tag_error, tag), Toast.LENGTH_LONG).show()
+            finish()
+        }
+
+        supportActionBar?.title = tag
+
+        viewModel.refreshing.observe(this) {
+            refreshLayout?.isRefreshing = it
```
Comment (brittanyberlanga) [inline app/src/main/java/me/tylerbwong/stack/ui/questions/tags/SingleTagQuestionsActivity.kt]: No need to null check

File: app/src/main/java/me/tylerbwong/stack/ui/questions/tags/SingleTagQuestionsViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        private val service: QuestionService = ServiceProvider.questionService
+) : BaseViewModel() {
+
+    internal val data: LiveData<List<DynamicDataModel>>
+        get() = _data
+    private val _data = MutableLiveData<List<DynamicDataModel>>()
+
+    internal var currentTag: String = ""
+    @Sort
+    internal var currentSort: String = CREATION
```
Comment (brittanyberlanga) [inline app/src/main/java/me/tylerbwong/stack/ui/questions/tags/SingleTagQuestionsViewModel.kt]: `currentTag` and `currentSort` can be private

File: app/src/main/java/me/tylerbwong/stack/data/DeepLinker.kt (hunk 1 above)
Comment (tylerbwong) [reply app/src/main/java/me/tylerbwong/stack/data/DeepLinker.kt]: Added an enum `ResolvedPath` to make this a little more type-safe 👍

Comment (brittanyberlanga) [reply app/src/main/java/me/tylerbwong/stack/data/DeepLinker.kt]: Cool, I like it! I was more so referring to the parts of the uri after those paths that you parse out.

e.g. QUESTIONS_BY_TAG expects `https://stackoverflow.com/questions/tagged/<tag>` and parses out `<tag>` 
whereas QUESTION_DETAILS expects `https://stackoverflow.com/questions/<questionId>/<questionTitle>` and parses out `<questionId>`

Comment (brittanyberlanga) [reply app/src/main/java/me/tylerbwong/stack/data/DeepLinker.kt]: I also noticed that some links that show up in search results follow the format 
`https://stackoverflow.com/questions/<questionId>/<questionTitle>/<id1>#<id2>`. Not 100% sure which one but one of the ids represents an answer id. Clicking the link takes you to the question and scrolls down to the answer. We could potentially add the same functionality in the future and just deep link to the question for now.

**Pull request merged**